pkg/agent/run.go

```diff
@@ -141,12 +140,12 @@ func Run(ctx context.Context, cfg cmds.Agent) error {
 		}
 	}
 
-	cfg.DataDir = filepath.Join(cfg.DataDir, "agent")
-	if err := os.MkdirAll(cfg.DataDir, 0700); err != nil {
+	agentDir := filepath.Join(cfg.DataDir, "agent")
+	if err := os.MkdirAll(agentDir, 0700); err != nil {
```

Comment:
we've replaced an in-place mutation with a local alias. the intentional side effect will no longer take place. this assumes that the side effect was intentional

Reply:
Correct. In my reading of the code the mutation to DataDir was incorrect; the only places that needed the modified paths have been updated to use the local var instead.